src/hostedPages/SchedulingActivity.stories.tsx

```diff
@@ -80,7 +94,39 @@ export const SchedulingActivity: Story = {
       };
     }, []);
 
-    const fetchProvidersFn = useCallback(() => args.fetchProviders(), []);
+    const [providerPrefs, setProviderPrefs] = useState<GetProvidersInputType>(
+      args.providerPreferences as GetProvidersInputType
+    );
+
+    const handlePrefsChange = (prefs: GetProvidersInputType) => {
+      console.log('Prefs changed', prefs);
+      setProviderPrefs(prefs);
+      fetchProvidersFn();
+    };
+
+    const fetchProvidersFn = useCallback(async () => {
+      const { data, ...rest } = await args.fetchProviders();
+      console.log('fetched providers');
+      return {
+        data: data.filter((p) => {
+          if (
+            providerPrefs.clinicalFocus &&
+            providerPrefs.clinicalFocus.length > 0
+          ) {
+            return some(
+              providerPrefs.clinicalFocus.map((f) =>
+                p.clinicalFocus
+                  ?.map((cf) => cf.toLowerCase())
+                  .includes(f.toLowerCase())
+              )
+            );
+          } else {
+            return true;
+          }
+        }),
+        ...rest
+      };
+    }, [providerPrefs]);
```

Comment:
Challenge: IMO this logic should live inside the `SchedulingActivity` component and not in the Story/Hosted Pages? I think this would require us to parameterize the `fetchProviders function so it includes the preferences to make that happen?

That way, we avoid business logic inside the story and Hosted Pages.

```typescript
const [providerPrefs, setProviderPrefs] = useState<GetProvidersInputType>(
  args.providerPreferences as GetProvidersInputType
);

// When preferences change, we force the component to call `fetchProviders` again
// This is logic we can add in SchedulingActivity.tsx
const handlePrefsChange = useCallback(() => args.onProviderPreferencesChange(), []);

// providerPrefs becomes a parameter of the fetch function
const fetchProvidersFn = useCallback((providerPrefs) => args.fetchProviders(providerPrefs), []);
```

Reply:
i wondered about this myself. we can actually reduce the number of props to the activity if we do that. so, 👍